libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/ThreadController.kt

```diff
@@ -95,10 +119,11 @@ class ThreadController {
     }
 
     /**
-     * Same as [getIOScopeAndRootJob], but using the MainThread dispatcher.
+     * Same as [getIOScopeAndRootJob], but using the SDK thread dispatcher.
      */
-    fun getMainScopeAndRootJob(): JobControl {
+    fun getSDKScopeAndRootJob(immediate: Boolean = false): JobControl {
```

Comment:
Will this flag help in all cases? What if someone invokes this method, keeps the scope as a val and uses `launch(Dispatchers.Main.Immediate)`? Or just `launch(Dispatchers.Main)`?
Since it's just for a test it's OK (we can check manually) but if it evolves into a constant solution I'd try to think of a way to disallow this behaviour in compile time.